driver-mapping/src/main/java/com/datastax/driver/mapping/AnnotationParser.java

```diff
@@ -21,263 +21,170 @@
 import com.datastax.driver.core.UserType;
 import com.datastax.driver.mapping.MethodMapper.ParamMapper;
 import com.datastax.driver.mapping.annotations.*;
+import com.google.common.annotations.VisibleForTesting;
 import com.google.common.base.Strings;
-import com.google.common.collect.Maps;
+import com.google.common.collect.ImmutableSet;
 import com.google.common.reflect.TypeToken;
 
+import java.beans.PropertyDescriptor;
 import java.lang.annotation.Annotation;
-import java.lang.reflect.Field;
-import java.lang.reflect.*;
+import java.lang.reflect.Method;
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
 import java.util.*;
 import java.util.concurrent.atomic.AtomicInteger;
 
 /**
- * Static metods that facilitates parsing class annotations into the corresponding {@link EntityMapper}.
+ * Static methods that facilitates parsing:
+ * - {@link #parseEntity(Class, MappingManager)}: entity classes into {@link EntityMapper} instances
+ * - {@link #parseUDT(Class, MappingManager)}: UDT classes into {@link MappedUDTCodec} instances.
+ * - {@link #parseAccessor(Class, MappingManager)}: Accessor interfaces into {@link AccessorMapper} instances.
  */
+@SuppressWarnings({"unchecked", "WeakerAccess"})
 class AnnotationParser {
 
-    private static final Comparator<Field> fieldComparator = new Comparator<Field>() {
+    @VisibleForTesting
+    static final Set<Class<? extends Annotation>> VALID_PROPERTY_ANNOTATIONS = ImmutableSet.of(
```

Comment:
Wasn't clear to me initially that VALID_FIELD_ANNOTATIONS apply to UDTs while PROPERTY apply to tables, would be good to clarify that here (although it may be somewhere else but haven't gotten to it yet).